src/authentication/SimpleCredentialsExtractor.ts

```diff
@@ -0,0 +1,16 @@
+import { Credentials } from './Credentials';
+import { CredentialsExtractor } from './CredentialsExtractor';
+import { HttpRequest } from '../server/HttpRequest';
+
+export class SimpleCredentialsExtractor extends CredentialsExtractor {
+  public async canHandle(): Promise<void> {
+    return undefined;
+  }
+
+  public async handle(input: HttpRequest): Promise<Credentials> {
+    if (input.headers.authorization) {
+      return { webID: input.headers.authorization };
+    }
+    return undefined;
```

Comment:
Is `undefined` a legal return value here? Should this not be a promise rejection?

Reply:
From what I understood this actor should almost never fail (except perhaps if there is a parsing problem in the header if it is present?) since having no identity is valid in a request. The AuthorizationHandler would fail if it gets no ID as input when one is required though.

Reply:
Ok regarding not having a promise rejection. Just wondering if `{}` should not be returned instead in that case.

Reply:
That would not be valid to the Credentials interface though, unless we make the fields there optional. If it is necessary to have something else than `undefined` an option could also be to have a static object representing no credentials being present.

Reply:
Ok, fair enough. So we need a falsy check in any case.

What I had in mind was that there could be multiple extractors producing hashes with some contents. Joining all of these hashes could then simply be a series of `Object.assign` if they are all guaranteed to not be undefined.
But no strong opinion on this.

Reply:
This might still change once Authorization/Authentication actually gets implemented because that section is still the vaguest one for me